Content.Shared/Drunk/DrunkSystem.cs

```diff
@@ -9,8 +11,33 @@ public abstract class SharedDrunkSystem : EntitySystem
     [ValidatePrototypeId<StatusEffectPrototype>]
     public const string DrunkKey = "Drunk";
 
-    [Dependency] private readonly StatusEffectsSystem _statusEffectsSystem = default!;
+    [ValidatePrototypeId<StatusEffectPrototype>]
+    public const string SleepKey = "ForcedSleep";
+    
+    [Dependency] protected readonly StatusEffectsSystem StatusEffectsSystem = default!;
     [Dependency] private readonly SharedSlurredSystem _slurredSystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<StatusEffectsComponent, StatusEffectEndedEvent>(OnDrunkEnded);
+    }
+
+    private void OnDrunkEnded(EntityUid uid, StatusEffectsComponent component, StatusEffectEndedEvent args)
+    {
+        if (args.Key == DrunkKey)
+        {
+            StatusEffectsSystem.TryRemoveStatusEffect(uid, SleepKey, component);
+            return;
+        }
+
+        if (args.Key == SleepKey)
+        {
+            StatusEffectsSystem.TryRemoveStatusEffect(uid, DrunkKey, component);
+            return;
+        }
```

Comment:
just combine the if-statements and pass the `args.Key` in the `TryRemoveStatusEffect`'s second argument

Reply:
but it doesn't remove status with args.Key, it removes the other status

Reply:
```suggestion
        if (args.Key == DrunkKey || args.Key == SleepKey)
        {
            StatusEffectsSystem.TryRemoveStatusEffect(uid, args.Key, component);
            return;
        }
```